### ISSUE TITLE
Feature: Snapshots

### ISSUE THREAD

**Issue by gronke** [iocage/libiocage#211]
Snapshots allow the user to zfs snapshot all data related to a jail.

### CLI Requirements

- [ ] Snapshots of a jail can be listed
  - [ ] Snapshot List is sorted by time of creation
- [ ] Snapshots can be deleted
- While a jail is running
  - [ ] Snapshots can be created
  - [ ] Snapshots can't be rolled back
- When a jail is stopped
  - [ ] Snapshots can be created
  - [ ] Snapshots can be rolled back

**Comment by gronke** [comment]
### Suggested CLI interface

```
# list snapshots of a jail
ioc snapshot jail

# create snapshot
ioc snapshot jail@snapshot

# delete snapshot
ioc snapshot remove jail@snapshot

# rollback jail to a previous snapshot
ioc snapshot rollback jail@snapshot
```

**Comment by gronke** [comment]
implemented in #246
